### PR TITLE
Upgrade minimatch, moment

### PR DESCRIPTION
**Pull request by rupal-bq**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-reports/package.json [hunk 1]
```diff
@@ -72,6 +72,9 @@
     "css-what": "^5.0.1",
     "ansi-regex": "5.0.1",
     "json-schema": "0.4.0",
-    "ws": "^7.4.6"
+    "ws": "^7.4.6",
+    "minimatch": "^3.0.5",
+    "terser": "^4.8.1",
+    "moment": "^2.29.4"
```
Comment (derek-ho) [inline dashboards-reports/package.json:78]: Were these used before? Where are they coming from?

**Review by derek-ho — APPROVED** [review]

File: dashboards-reports/package.json (hunk 1 above)
Comment (rupal-bq) [reply dashboards-reports/package.json:78]: Yes, all are existing dependencies. Just updating the version.

Comment (ps48) [reply dashboards-reports/package.json:78]: @rupal-bq can you please check if they are also present in OpenSearch-Dashboards? If so, we need not add the dependencies here again.

Comment (ps48) [reply dashboards-reports/package.json:78]: In Dashboards this is the current state in 2.x:
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/2.x/package.json
```
    "minimatch": "^3.0.4",
    "moment": "^2.24.0",

Comment (zhongnansu) [reply dashboards-reports/package.json:78]: fyi, in OSD, there's a PR pending to update `minimatch` to `3.0.5`, merged in main, pending approval for `2.x`
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2654/

Comment (rupal-bq) [reply dashboards-reports/package.json:78]: Thanks @ps48, I didn't know that. Some are present in OpenSearch-Dashboards. I'll update PR accordingly.

**Comment by rupal-bq** [comment]
Added changes to https://github.com/opensearch-project/dashboards-reports/pull/515

File: dashboards-reports/package.json (hunk 1 above)
Comment (joshuali925) [reply dashboards-reports/package.json:78]: does dashboards resolving it to 3.0.5 prevent plugins resolving it to 3.0.4? I think Dashboards package.json direct dependency will not affect transitive dependencies?

**Comment by rupal-bq** [comment]
Reopening PR since Opensearch Dashboards 2.x has minimatch v3.0.5 but reporting still has v3.0.4

**Review by joshuali925 — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

**Pull request merged**